tests/functional/api.test.js

```diff
@@ -353,7 +354,29 @@ testDriver.test('api is available when sessionStorage is not', function (t, brow
     .catch(fail(t))
 })
 
-testDriver.test('setUserId adds correct attribute to jserror', function (t, browser, router) {
+testDriver.test('setCustomAttribute can persist onto subsequent page loads', notIE, function (t, browser, router) {
+  let loadPromise = browser.get(router.assetURL('instrumented.html', { scriptString: 'newrelic.setCustomAttribute(\'testing\',123,true);' }))
+  Promise.all([router.expectRum(), loadPromise])
+    .then(([{ request: { query } }]) => {
+      t.equal(query.ja, '{"testing":123}', 'initial page load has custom attribute')
+
+      loadPromise = browser.get(router.assetURL('instrumented.html'))
+    })
+    .then(() => Promise.all([router.expectRum(), loadPromise])) // testing:123 is still expected on next page load within same tab
+    .then(([{ request: { query } }]) => {
+      t.equal(query.ja, '{"testing":"123"}', '2nd page load still has custom attribute gotten from storage (but converted to string)')
+
+      loadPromise = browser.get(router.assetURL('instrumented.html', { scriptString: 'newrelic.setCustomAttribute(\'testing\',null);' }))
+    })
+    .then(() => Promise.all([router.expectRum(), loadPromise])) // testing should've been erased from storage
+    .then(([{ request: { query } }]) => {
+      t.equal(query.ja, undefined, '3rd page load does not retain custom attribute after unsetting (set to null)')
+      t.end()
+    })
+    .catch(fail(t))
+})
+
+testDriver.test('setUserId adds correct (persisted) attribute to payloads', withUnload.and(notIE), function (t, browser, router) {
```

Comment:
Just for the record, does this not work with IE11?

Reply:
sessionStorage should work for IE so I am curious

Reply:
It couldn't functionally retrieve it from sessionStorage on `browser.refresh()`. Uncertain if it's a WD/Sauce issue or the api, but since IE still otherwise work, I kicked it aside.